### PR TITLE
icu4c.rb: Correct the SVN repository URL

### LINKED ISSUE #7249: icu4c.rb: Head repo URL incorrect
`"https://ssl.icu-project.org/repos/icu/icu/trunk/", :using => :svn` does not exist. Best I can tell is that it has changed to `https://ssl.icu-project.org/repos/icu/trunk/icu4c/`, It doesn't look like icu4j has a formula, so this should be the only change.

### PR DESCRIPTION
**Pull request by moderndeveloperllc**
This fixes #7249.

### CODE REVIEW

**Comment by woodruffw** [comment]
```
==> /usr/local/Cellar/mdk/1.2.9_1/bin/mixasm hello
hello.mixal:12: Warnung: Operand von ALF-Pseudoinstruktion hat weniger als 5 Zeichen
hello.mixal:0: Warnung: Symbol nicht definiert: halt
(2 Warnung(en))
Error: mdk: failed
<"Program loaded. Start address: 1000\nRunning ...\nMIXAL HELLO WORLD                                                     \n... done\n"> expected but was
<"Programm geladen.  Startadresse: 1000\nArbeite ...\nMIXAL HELLO WORLD                                                     \n... fertig\n">.
```

Very strange. Looks like we're hitting some accidental internationalization somewhere, and only on the Sierra CI.

**Comment by woodruffw** [comment]
@BrewTestBot test this please

**Comment by moderndeveloperllc** [comment]
@woodruffw New to Homebrew PRs. Is this something I need to fix, or is this a tooling issue?

**Comment by woodruffw** [comment]
I don't believe the failure's being caused by the new `head`, so it's probably a tooling/CI issue. I'll look into this.

**Comment by DomT4** [comment]
Hey it's the German test again @ilovezfs 😸

**Comment by woodruffw** [comment]
@moderndeveloperllc For the time being, let's just add `ENV["LANG"] = "en_US.UTF-8"` to the top of the `test` block explaining that it's a CI discrepancy.

**Comment by ilovezfs** [comment]
```
==> brew test --verbose mdk
==> FAILED
Testing mdk
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20161130-89361-4zaxi.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mdk.rb --verbose
==> /usr/local/Cellar/mdk/1.2.9_1/bin/mixasm hello
hello.mixal:12: Warnung: Operand von ALF-Pseudoinstruktion hat weniger als 5 Zeichen
hello.mixal:0: Warnung: Symbol nicht definiert: halt
(2 Warnung(en))
Error: mdk: failed
<"Program loaded. Start address: 1000\nRunning ...\nMIXAL HELLO WORLD                                                     \n... done\n"> expected but was
<"Programm geladen.  Startadresse: 1000\nArbeite ...\nMIXAL HELLO WORLD                                                     \n... 
```

it's mdk not icu4c that needs the language fix

File: Formula/icu4c.rb [hunk 1]
```diff
@@ -35,6 +35,8 @@ def install
   end
 
   test do
+    # CI dependancy
```
Comment (MikeMcQuaid) [inline Formula/icu4c.rb]: Remove this comment.

File: Formula/icu4c.rb [hunk 2]
```diff
@@ -35,6 +35,8 @@ def install
...
+    ENV["LANG"] = "en_US.UTF-8"
```
Comment (MikeMcQuaid) [inline Formula/icu4c.rb]: If I understood @ilovezfs correctly this line isn't needed.

File: Formula/mdk.rb [hunk 3]
```diff
@@ -30,6 +30,8 @@ def install
   end
 
   test do
+    # CI dependancy
```
Comment (MikeMcQuaid) [inline Formula/mdk.rb]: Remove this comment.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by woodruffw** [comment]
@moderndeveloperllc The mdk fix is probably best isolated to a separate PR (I opened #7502 for it), so you can pop that commit off.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `icu4c: correct the HEAD SVN repository URL.`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Comment by MikeMcQuaid** [comment]
Merged in 5374bd5ab2f700f7b0f121f67d96636365bd5521

**Pull request closed without merging**